### ISSUE TITLE
Pyscripter 3.6.4 generates "Out of memory" error when called / opened from another application

### ISSUE THREAD

**Issue by mboeringa** [pyscripter/pyscripter#1035]
Hi,

After upgrading to Pyscripter 3.6.4, I am no longer able to open Pyscripter from within ESRI's ArcGIS (Geographical Information System). ArcGIS (ArcMap in this particular case), has the ability to set a prefered Python editor, and to start an edit session of a Python script by launching Pyscripter from within the application.

While this still works for "stand-alone" Python scripts present as *.py files on the file system, this now fails with an "Out of memory" type error message thrown by Pyscripter when it launches for scripts embedded in ArcGIS binary toolboxes, that normally get extracted to a temporary *.py file by ArcGIS before launching Pyscripter. It is this temporary file that is actually being edited. After saving it, the handle goes back to ArcGIS.

I don't know at what stage of process this is actually failing, but after downgrading to 3.6.3 it works again (and has been fully functional for the past 10 years or so with older versions of ArcGIS and Pyscripter).

**Comment by pyscripter** [comment]
Does "Edit with PyScripter" from the context menu of the File Explorer work?

**Comment by amr66** [comment]
Same here, Pyscripter is installed from zip, it is the 32-bit version, calling it from a batch file. Windows is Win10, 64bit

    Error in opening file:
    "Path_to_...PyScripter.exe"
    Error: Out of memory

Strange, if I call the 64-bit version from batch, that works.

**Comment by pyscripter** [comment]
@amr66 This is helpful.
@mboeringa Are you also using the 32 bit version?

**Comment by amr66** [comment]
mboeringa said, he uses ArcMap, so, yes it is 32bit ([ESRI about python version for Desktop](https://support.esri.com/en/technical-article/000013224))

**Comment by mboeringa** [comment]
@pyscripter

Sorry for the scandalously late response and follow up on the issue I started. I got swamped by other stuff and didn't get around to this.

Yes, as @amr66 confirmed, I am using the 32bit version as that is indeed the limitation of ArcMap, as still being a 32bit app.

> Does "Edit with PyScripter" from the context menu of the File Explorer work?

Simply right clicking the file and choosing "Open" in the Windows Explorer, does work without issues and scripts are opened fine. I do not see a specific "Edit with PyScripter" context menu option listed though in Windows Explorer, even though I chose to install all options like "Shortcut on desktop" and such.

I have been using 3.6.3 for the past months due to this issue, but now re-installed 3.6.4 to see if the issue persisted over the recent Windows updates and such. It is still a problem, same "Out of memory" error:

![afbeelding](https://user-images.githubusercontent.com/7635726/109415882-90326880-79bb-11eb-8011-5527d31885f7.png)
